plugin/pkg/admission/resourcequota/admission.go

```diff
@@ -59,213 +68,191 @@ func NewResourceQuota(client clientset.Interface) admission.Interface {
-			delta, err := resourcequotacontroller.PodRequests(pod, resourceName)
-
+			_, err = q.client.Core().ResourceQuotas(quotaToUpdate.Namespace).UpdateStatus(quotaToUpdate)
 			if err != nil {
-				return false, fmt.Errorf("%s is limited by quota, must make explicit request.", resourceName)
-			}
-
-			// if this operation is an update, we need to find the delta usage from the previous state
-			if a.GetOperation() == admission.Update {
-				oldPod, err := client.Core().Pods(a.GetNamespace()).Get(pod.Name)
-				if err != nil {
-					return false, err
+				if !errors.IsConflict(err) {
+					return admission.NewForbidden(a, fmt.Errorf("Unable to update quota status: %s %v", resourceQuota.Name, err))
 				}
-
-				// if the previous version of the resource made a resource request, we need to subtract the old request
-				// from the current to get the actual resource request delta.  if the previous version of the pod
-				// made no request on the resource, then we get an err value.  we ignore the err value, and delta
-				// will just be equal to the total resource request on the pod since there is nothing to subtract.
-				oldRequest, err := resourcequotacontroller.PodRequests(oldPod, resourceName)
-				if err == nil {
-					err = delta.Sub(*oldRequest)
-					if err != nil {
-						return false, err
-					}
+				latestQuota, err := q.client.Core().ResourceQuotas(namespace).Get(resourceQuota.Name)
+				if err != nil {
+					return admission.NewForbidden(a, fmt.Errorf("Unable to get quota: %s %v", resourceQuota.Name, err))
 				}
+				tryAgain = append(tryAgain, latestQuota)
 			}
+		}
 
-			newUsage := used.Copy()
-			newUsage.Add(*delta)
+		// all quotas were updated, so we can return
+		if len(tryAgain) == 0 {
+			return nil
+		}
 
-			// make the most precise comparison possible
-			newUsageValue := newUsage.Value()
-			hardUsageValue := hard.Value()
-			if newUsageValue <= resource.MaxMilliValue && hardUsageValue <= resource.MaxMilliValue {
-				newUsageValue = newUsage.MilliValue()
-				hardUsageValue = hard.MilliValue()
+		// we have concurrent requests to update quota, so look to retry if needed
+		// next iteration, we need to process the items that have to try again
+		// pause the specified interval to encourage jitter
+		if retry == numRetries {
+			names := []string{}
+			for _, quota := range tryAgain {
+				names = append(names, quota.Name)
 			}
+			return admission.NewForbidden(a, fmt.Errorf("Unable to update status for quota: %s, ", strings.Join(names, ",")))
+		}
+		resourceQuotasToProcess = tryAgain
+		time.Sleep(interval)
+	}
+	return nil
+}
 
-			if newUsageValue > hardUsageValue {
-				errs = append(errs, fmt.Errorf("%s quota is %s, current usage is %s, requesting %s.", resourceName, hard.String(), used.String(), delta.String()))
-				dirty = false
-			} else {
-				status.Used[resourceName] = *newUsage
-			}
+// prettyPrint formats a resource list for usage in errors
+func prettyPrint(item api.ResourceList) string {
```

Comment:
Taking back. Let's keep the precision. We can re-render the values during event generation if needed.